### PR TITLE
Snapshot only specific files for COPY

### LINKED ISSUE #314: Image sizes increased dramatically since merging #289
Since merging #289, Kaniko seems to create layers containing not only the files copied by `COPY` instruction, but also files from the previous layers.

### PR DESCRIPTION
**Pull request by priyawadhwa**
Before #289 was merged, when copying over directories for COPY kaniko
would get a list of all files at the destination specified and add them
to the list of files to be snapshotted. If the destination was root it
would add all files. This worked because the snapshotter made sure the
file had been changed before adding it to the layer.

After #289, we changed the logic to add all files snapshotted to a layer
without checking if the files had been changed. This created the bug in which kaniko
got all the files at root and added them to the layer without checking
if they had been changed.

This change should fix this bug. Now, the CopyDir function returns a
list of files it copied over and only those files are added to the list
of files to be snapshotted.

I also added this change to the `UnpackLocalTarArchive` function in ADD, so that we only add specific files extracted from local tar archives to the list of snapshotted files.

Should fix #314

Once #317 is implemented, this change will be tested by integration tests. Until then, I locally tested the Dockerfile in the bug and it seems to be working now.

### CODE REVIEW

File: pkg/util/fs_util.go [hunk 1]
```diff
@@ -155,21 +155,24 @@ func ChildDirInWhitelist(path, directory string) bool {
 	return false
 }
 
-func unTar(r io.Reader, dest string) error {
+// unTar returns a list of files extracted from the tar archive
```
Comment (bobcatfish) [inline pkg/util/fs_util.go]: would be good to including in this docstring what the params `r` and `dest` are for (https://blog.golang.org/godoc-documenting-go-code) e.g. `unTar returns a list of files that have been extracted from the tar archive at r to the path at dest`

**Comment by bobcatfish** [comment]
I'm sorry that that dog looks so suspicious

File: pkg/util/fs_util.go (hunk 1 above)
Comment (priyawadhwa) [reply pkg/util/fs_util.go]: for sure, just added it!

**Pull request merged**